### DISCUSSION TITLE
Nearest Neighbor interpolation?

### DISCUSSION THREAD

**Discussion by blue-j** [ImageMagick/ImageMagick#3955] [Help] [answered]
I'm trying to get a nearest neighbor interpolated resize going, and am not successful.

I want this source image:
![test-cluster-5](https://user-images.githubusercontent.com/3990142/126057628-afea5f2f-f9b0-4682-8575-64789c6252b3.jpg)

to end up like this, resized 1000%, with nearest neighbor interpolation, except as a PGM (not an accepted format here):
![test-cluster-5_resized](https://user-images.githubusercontent.com/3990142/126057656-dfe71c1e-e694-4d0d-88ca-f8966dd114b5.jpg)

but when I try
convert input.jpg -interpolate nearest -resize 1000% output.pgm

I get the result I'd expect from a bilinear or bicubic method, blurred and not pixelly and sharp as desired with nearest neighbor.

Any help solving this would be most appreciated!  : )

J

**Comment by snibgo** [comment]
What version of IM, on what platform? If you use IM v7, I suggest "magick" instead of "convert".

"-resize" is not sensitive "-interpolate" (see https://imagemagick.org/script/command-line-options.php#interpolate), so all the "-interpolate" methods will give the same result.

"-resize" is sensitive to "-filter" (see https://imagemagick.org/script/command-line-options.php#filter). Use "-filter box" or "-filter point" before the "-resize.

However, for the effect you want, I suggest "-scale" which is much faster.
```
magick in.png -scale 1000% out.pgm
```

**Comment by fmw42** [comment]
try adding -filter point to your command

`convert input.jpg -interpolate nearest -filter point -resize 1000% output.pgm`

**Comment by snibgo** [comment]
With "-resize", there is no point in using "-interpolate". It does nothing, and using it may confuse human readers.

**Reply by fmw42** [reply]
Anthony explained to me (assuming I understood), that adding -filter point disabled the filter in -resize so that -interpolate could be added as

`convert input.jpg -interpolate nearest -filter point -resize 1000% output.pgm`

**Comment by fmw42** [comment, marked as answer]
See -interpolative-resize at https://legacy.imagemagick.org/Usage/resize/#interpolative-resize

**Reply by snibgo** [reply]
Thanks, I didn't know about "-interpolative-resize". It isn't documented on https://imagemagick.org/script/command-line-options.php . Could this be added, please?

**Reply by fmw42** [reply]
I have requested it also.

**Reply by fmw42** [reply]
It is listed now.

**Comment by blue-j** [comment]
Thanks so much to both of you!  My pixel Jedi.  Deep bow.  : )

The interpolative-resize with nearest is what brought full joy.  - J

**Reply by blue-j** [reply]
I should add that the command ONLY works with magick.  If you use convert it says:

convert: unrecognized option `-interpolative-resize' @ error/convert.c/ConvertImageCommand/1952.

**Reply by fmw42** [reply]
Works fine for me on IM 6.9.12.19 and 7.1.0.4 Mac OSX Sierra as

`convert lena.png -interpolative-resize 50% x.png`  on IM 6

`magick lena.png -interpolative-resize 50% x.png`  on IM 7

**Reply by blue-j** [reply]
I included the argument

"-interpolate nearest" before the "-interpolative-resize"!

Is that not what was suggested?  Works fine with magick. 

- J

**Reply by fmw42** [reply]
Works fine for me with -interpolate nearest included in the command on both IM 6 and IM 7.

This works for me on IM 6.

`convert lena.png -interpolate nearest -interpolative-resize 50% x.png`

**Reply by blue-j** [reply]
Tried again with your exact command, same result.  with convert I get:
convert: unrecognized option `-interpolative-resize' @ error/convert.c/ConvertImageCommand/1952

With magick, it works.  Using 7.1.0-4 Q16 x86_64 on Big Sur (11.4).  Maybe it is OS related somehow?  I'm going to uninstall and reinstall and see what that does.   - J

**Comment by blue-j** [comment]
I owe you this info:

Version: ImageMagick 7.0.11-13 Q16 x86_64 2021-05-17 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): bzlib cairo djvu fftw fontconfig freetype heic jbig jng jpeg lcms lqr lzma openexr pangocairo png rsvg tiff webp wmf x xml zlib
